server/schedule/filter/filters.go

```diff
@@ -295,66 +295,126 @@ func (f *StoreStateFilter) Type() string {
 type conditionFunc func(*config.PersistOptions, *core.StoreInfo) bool
 
 func (f *StoreStateFilter) isRemoved(opt *config.PersistOptions, store *core.StoreInfo) bool {
-	f.Reason = "tombstone"
-	return store.IsRemoved()
+	isRemoved := store.IsRemoved()
```

Comment:
hou about:
```suggestion
        f.Reason = ""
	isRemoved := store.IsRemoved()
	if isRemoved {
	      f.Reason = "tombstone"
	}
	return isRemoved
```
others also can change like it.

Reply:
Thanks for reviewing!   I think both can get the same result, use if{}else{}  Reason will be set  only once, but set reason and use if{}, Reason may be set twice,  which may have a little influence on the performance.